### PR TITLE
rpc: fix tx_search pagination with ordered results

### LINKED ISSUE #4432: rpc: tx search pagination duplicates
@alexanderbez says:

> txs search pagination seems to have regressed
>do 2 queries:
>`page=1&per_page=5`
>`page=2&per_page=5`
>you'll get dup txs in both results

### PR DESCRIPTION
**Pull request by erikgrinaker**
Fixes #4432.

* [x] Referenced an issue explaining the need for the change
* [x] ~Updated all relevant documentation in docs~
* [x] Updated all code comments where relevant
* [x] Wrote tests
* [x] Updated CHANGELOG_PENDING.md

### CODE REVIEW

File: rpc/client/rpc_test.go [hunk 1]
```diff
@@ -414,14 +415,25 @@ func TestTx(t *testing.T) {
...
-	// first we broadcast a tx
 	c := getHTTPClient()
-	_, _, tx := MakeTxKV()
-	bres, err := c.BroadcastTxCommit(tx)
-	require.Nil(t, err)
 
-	txHeight := bres.Height
-	txHash := bres.Hash
+	// first we broadcast a few txs
+	var tx []byte
```
Comment (melekes) [inline rpc/client/rpc_test.go]: small nit:

```
var (
)
```

File: rpc/client/rpc_test.go [hunk 2]
```diff
@@ -495,6 +503,31 @@ func TestTxSearch(t *testing.T) {
 			require.GreaterOrEqual(t, result.Txs[k].Height, result.Txs[k+1].Height)
 			require.GreaterOrEqual(t, result.Txs[k].Index, result.Txs[k+1].Index)
 		}
+
+		// check pagination
+		seen := map[int64]bool{}
```
Comment (melekes) [inline rpc/client/rpc_test.go]: var (
)

File: rpc/client/rpc_test.go [hunk 3]
```diff
@@ -495,6 +503,31 @@ func TestTxSearch(t *testing.T) {
...
+		maxHeight := int64(0)
+		perPage := 3
+		pages := txCount/perPage + 1
```
Comment (melekes) [inline rpc/client/rpc_test.go]: math.Ceil or create a function and use it here

File: rpc/core/tx.go [hunk 4]
```diff
@@ -72,56 +72,54 @@ func TxSearch(ctx *rpctypes.Context, query string, prove bool, page, perPage int
...
-		index := r.Index
+	apiResults := make([]*ctypes.ResultTx, 0, pageSize)
+	for i := skipCount; i < skipCount+pageSize; i++ {
+		r := results[i]
 
+		var proof types.TxProof
 		if prove {
-			block := blockStore.LoadBlock(height)
-			proof = block.Data.Txs.Proof(int(index)) // XXX: overflow on 32-bit machines
+			block := blockStore.LoadBlock(r.Height)
```
Comment (melekes) [inline rpc/core/tx.go]: let's open an issue to cache blocks

Comment (marbar3778) [reply rpc/core/tx.go]: would this fall under: https://github.com/tendermint/tendermint/issues/1770

Comment (melekes) [reply rpc/core/tx.go]: it's similar

File: rpc/client/rpc_test.go [hunk 5]
```diff
@@ -414,14 +415,25 @@ func TestTx(t *testing.T) {
...
-	txHash := bres.Hash
+	// first we broadcast a few txs
+	var tx []byte
+	var txHeight int64
+	var txHash tmbytes.HexBytes
+	for i := 0; i < 10; i++ {
+		_, _, tx = MakeTxKV()
+		res, err := c.BroadcastTxCommit(tx)
+		require.NoError(t, err)
+		txHeight = res.Height
```
[the remark this answers is not in the document]
Comment (erikgrinaker) [reply rpc/client/rpc_test.go]: Fixed by #4446, along with other nits

File: rpc/client/rpc_test.go (hunk 1 above)
Comment (erikgrinaker) [reply rpc/client/rpc_test.go]: This should really be enforced by a linter.

**Pull request merged**